### PR TITLE
(#2) - introduce assert-that object

### PR DESCRIPTION
**Pull request by Graur**

### CODE REVIEW

**Comment by Graur** [comment]
This pull request is in Work In Progress status, for now.

**Comment by yegor256** [comment]
@Graur I think that `equal-to` doesn't need to be an atom, it can easily be implemented in EO

**Comment by Graur** [comment]
> @Graur I think that `equal-to` doesn't need to be an atom, it can easily be implemented in EO

It does. But I didn't add array comparisons logic. This will not work:

```
[] > two-arrays
  assert-that > @
    * 1 2 3
    $.equal-to * 1 2 3
    "two array"
```

I've got plan to implement it in my next PR

Something like this: 
``` 
if (dataizedActual.getClass().isArray()) {
                return new Data.ToPhi(dataizedExpected.getClass().isArray() && Arrays.equals(((Object[]) dataizedActual), ((Object[])dataizedExpected)));
            }
```

**Comment by Graur** [comment]
> @Graur I think that `equal-to` doesn't need to be an atom, it can easily be implemented in EO

After our discussion I remove all .java files

**Comment by Graur** [comment]
@yegor256 check again, please

File: src/main/eo/org/eolang/hamcrest/assert-that.eo [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
++alias sprintf org.eolang.txt.sprintf
++rt jvm org.eolang:eo-hamcrest:0.0.0
+
+# Main object for assertions
+[actual matcher reasons...] > assert-that
+
+  if. > @
+    matcher.match actual
+    TRUE
+    []
```
Comment (yegor256) [inline src/main/eo/org/eolang/hamcrest/assert-that.eo:33]: @Graur you can replace it with just `sprintf`. Why making a new abstract object `[]`?

File: src/main/eo/org/eolang/hamcrest/assert-that.eo [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    [a] > match
+      eq. > @
+        x
+        a
+
+    [] > describe
+      sprintf > @
+        "was <%s>"
+        ^.x
```
Comment (yegor256) [inline src/main/eo/org/eolang/hamcrest/assert-that.eo:54]: @Graur you don't need `^` here, just `x` is enough

**Comment by yegor256** [comment]
@Graur done, see above

File: src/main/eo/org/eolang/hamcrest/assert-that.eo (hunk 1 above)
Comment (Graur) [reply src/main/eo/org/eolang/hamcrest/assert-that.eo]: Seems to me more readable, but I agree, we can remove it

File: src/main/eo/org/eolang/hamcrest/assert-that.eo [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+[actual matcher reasons...] > assert-that
+
+  if. > @
+    matcher.match actual
+    TRUE
+    []
+      sprintf > @
+        "%s Expected: <%s> but %s"
+        if.
+          &.reasons.is-empty
```
[the remark this answers is not in the document]
Comment (Graur) [reply src/main/eo/org/eolang/hamcrest/assert-that.eo]: It works without `&`, thanks.

File: src/main/eo/org/eolang/hamcrest/assert-that.eo (hunk 2 above)
Comment (Graur) [reply src/main/eo/org/eolang/hamcrest/assert-that.eo]: Agree, fixed

**Comment by Graur** [comment]
@yegor256 I removed all unnecessary objects. Could you check it again?

**Comment by yegor256** [comment]
@Graur looks good

**Pull request merged**